src/main/java/org/jetbrains/bsp/bazel/server/bazel/BazelRunnerBuilder.java

```diff
@@ -0,0 +1,68 @@
+package org.jetbrains.bsp.bazel.server.bazel;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.bsp.bazel.server.bazel.data.ProcessResults;
+import org.jetbrains.bsp.bazel.server.bazel.utils.BazelArgumentsUtils;
+
+public class BazelRunnerBuilder {
+
+  private final BazelRunner bazelRunner;
+  private final String bazelCommand;
+
+  private final List<String> flags;
+  private final List<String> arguments;
+
+  BazelRunnerBuilder(BazelRunner bazelRunner, String bazelCommand) {
+    this.bazelRunner = bazelRunner;
+    this.bazelCommand = bazelCommand;
+    this.flags = new ArrayList<>();
+    this.arguments = new ArrayList<>();
+  }
+
+  public BazelRunnerBuilder withFlag(BazelRunnerFlag flag) {
+    flags.add(flag.toString());
+
+    return this;
+  }
+
+  public BazelRunnerBuilder withFlag(BazelRunnerFlag flag, String value) {
```

Comment:
Perhaps we could create methods for very common flags, especially the proto output one.

Reply:
I think that it is unnecessary - I would like to keep builder as simple as possible. Nevertheless `.withProtoOutputFlag` call is almost the same sa `.withFlag(PROTO_OUTPUT_FLAG)`

Reply:
I think it would provide a simpler api from who will handle it, since you don't have to worry about the flags and where the imports are to them are. However, I can see that this might bring out some ambiguity to the calls. Maybe a third opinion would be interesting, like @agluszak 's

Reply:
My initial thought was to have the builder for custom calls and create methods for the most common uses, but after a closer look there aren't any places where the builder is used with the same params more then once :D I agree with Marcin that the builder itself should be kept simple

Reply:
Alright let's leave it as is :smile: